### PR TITLE
View trips page

### PR DESCRIPTION
**Pull request by GeoM-bit**

### CODE REVIEW

File: src/app/core/services/authentication.service.ts [hunk 1]
```diff
@@ -21,9 +21,10 @@ export class AuthenticationService {
     return this.http.post<TokenModel>(environment.baseUrl + '/api/user/login', user);
   }
 
-  getUserRole(response: TokenModel)
+  getUserRole()
```
Comment (paulcrasmareanu) [inline src/app/core/services/authentication.service.ts:24]: please add return type,

File: src/app/features/components/filter-form/filter-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  clearStartDate: Date=null;
+  searchCriteria = new FilterModel();
+  statusTypes = Object.values(RequestStatus).filter(value=>typeof value === 'number');
+  confirmValidDateMatcher = new ConfirmValidDateMatcher();
+  missingDateMatcher = new MissingDateMatcher();
+
+  @Output()
+  filter = new EventEmitter<FilterModel>();
+
+  constructor(private dateAdapter: DateAdapter<Date>, private _liveAnnouncer: LiveAnnouncer, private viewTripsService: ViewTripsService, private authenticationService: AuthenticationService) {
```
Comment (paulcrasmareanu) [inline src/app/features/components/filter-form/filter-form.component.ts:33]: please remove _liveAnnouncer

File: src/app/features/components/filter-form/filter-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    this.checkForWhiteSpaces();
+    this.filterComplete(this.searchCriteria);
+  }
+
+  onReset(){
+    this.filterComplete(new FilterModel());
+  }
+
+  initForm() {
+    if (this.authenticationService.getUserRole() == UserRoles[0]) {
```
Comment (paulcrasmareanu) [inline src/app/features/components/filter-form/filter-form.component.ts:53]: Please use an enum for UserRoles and rename in Roles.

File: src/app/features/components/login/login.component.ts [hunk 4]
```diff
@@ -32,21 +32,19 @@ export class LoginComponent implements OnInit {
     this.loginModel = this.loginForm.value;
     this.authenticationService.login(this.loginModel).subscribe((response: TokenModel) => {
       if(response!=null) {
-        this.redirectToViewBusinessTripsBasedOnRole(response);
+        localStorage.setItem('token', JSON.stringify({ token: response.token}));
```
Comment (paulcrasmareanu) [inline src/app/features/components/login/login.component.ts:35]: please remove this, you should use that in interceptor

File: src/app/features/components/trips-table/trips-table.component.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+})
+
+export class TripsTableComponent implements OnInit, AfterViewInit{
+  displayedColumnsUser: string[] = ['client', 'clientLocation', 'projectName', 'startDate', 'endDate', 'accommodation', 'status', 'action'];
+  displayedColumnsBTO: string[] = ['firstName', 'email','pmName','area','client', 'clientLocation', 'projectName', 'startDate', 'endDate', 'accommodation', 'action'];
+  displayedColumns: string[];
+  status: RequestStatus;
+  statusMapping = StatusMapping;
+  areaMapping = AreaMapping;
+  user: String;
```
Comment (paulcrasmareanu) [inline src/app/features/components/trips-table/trips-table.component.ts:33]: please rename this, string

File: src/app/features/components/trips-table/trips-table.component.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  userRole = UserRoles[0];
```
Comment (paulcrasmareanu) [inline src/app/features/components/trips-table/trips-table.component.ts:34]: Please use a method for roles

File: src/app/features/components/trips-table/trips-table.component.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        this.dataSource.sort=this.sort;
+        if(this.dataSource.data.length==0) {
+          this.noTrips = true;
+        }
+      })
+    }
+    this.noTrips = this.dataSource.data.length==0 ? true:false;
+  }
+
+  checkBtoRole(){
```
Comment (paulcrasmareanu) [inline src/app/features/components/trips-table/trips-table.component.ts:86]: please remove.

**Review by paulcrasmareanu — CHANGES_REQUESTED** [review]

File: src/app/core/services/authentication.service.ts (hunk 1 above)
Comment (GeoM-bit) [reply src/app/core/services/authentication.service.ts]: Done

File: src/app/features/components/filter-form/filter-form.component.ts (hunk 3 above)
Comment (GeoM-bit) [reply src/app/features/components/filter-form/filter-form.component.ts]: Done

File: src/app/features/components/login/login.component.ts (hunk 4 above)
Comment (GeoM-bit) [reply src/app/features/components/login/login.component.ts:35]: It proves that I need to set this here in order to decode the role of the user later; the interceptor will only read it after each request and add it to the header.

File: src/app/features/components/trips-table/trips-table.component.ts (hunk 5 above)
Comment (GeoM-bit) [reply src/app/features/components/trips-table/trips-table.component.ts]: Done

File: src/app/features/components/trips-table/trips-table.component.ts (hunk 7 above)
Comment (GeoM-bit) [reply src/app/features/components/trips-table/trips-table.component.ts]: Done

File: src/app/features/components/filter-form/filter-form.component.ts (hunk 2 above)
Comment (GeoM-bit) [reply src/app/features/components/filter-form/filter-form.component.ts]: Done

File: src/app/features/components/trips-table/trips-table.component.ts (hunk 6 above)
Comment (GeoM-bit) [reply src/app/features/components/trips-table/trips-table.component.ts]: Done

File: src/app/features/components/login/login.component.ts (hunk 4 above)
Comment (paulcrasmareanu) [reply src/app/features/components/login/login.component.ts:35]: You can use a service to receive the role directly.

**Review by paulcrasmareanu — CHANGES_REQUESTED** [review]

**Review by paulcrasmareanu — APPROVED** [review]

**Pull request merged**